cpp/velox/substrait/SubstraitToVeloxExpr.cc

```diff
@@ -163,6 +180,44 @@ VectorPtr constructFlatVector(
   return vector;
 }
 
+TypePtr getScalarType(const ::substrait::Expression::Literal& literal) {
+  auto typeCase = literal.literal_type_case();
+  switch (typeCase) {
+    case ::substrait::Expression_Literal::LiteralTypeCase::kBoolean:
+      return BOOLEAN();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kI8:
+      return TINYINT();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kI16:
+      return SMALLINT();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kI32:
+      return INTEGER();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kFp32:
+      return REAL();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kI64:
+      return BIGINT();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kFp64:
+      return DOUBLE();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kString:
+      return VARCHAR();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kDate:
+      return DATE();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kTimestamp:
+      return TIMESTAMP();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kVarChar:
+      return VARCHAR();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kBinary:
+      return VARBINARY();
+    case ::substrait::Expression_Literal::LiteralTypeCase::kDecimal: {
+      auto precision = literal.decimal().precision();
+      auto scale = literal.decimal().scale();
+      auto type = DECIMAL(precision, scale);
+      return type;
+    }
+    default:
+      return nullptr;
```

Comment:
Does it make sense to throw exception here? For now the caller needs to throw for unsupported case anyway. After the change it doesn't need to check and handle invalid return value.

Reply:
Now we have 3 callers for this function, which throw different error msgs for the invalid return value. As this function only covers scalar types, I think this is better to let callers to handle the invalid return value.